TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -10,40 +10,61 @@ import Foundation
 class NetworkManager {
     
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
-        guard var components = URLComponents(string: urlString) else {
-            completion(nil)
-            return
-        }
-        components.queryItems = parameters.map { (key, value) in
-            URLQueryItem(name: key, value: value)
-        }
-        components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
-        guard let url = components.url else {
-            completion(nil)
-            return
-        }
-        var urlRequest = URLRequest(url: url)
-        urlRequest.httpMethod = "GET"
-        let urlSession = URLSession.shared
-        let fileDownloadTask = urlSession.dataTask(
-            with: urlRequest
-        ) { data, _, error in
-            let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
-                DispatchQueue.main.async {
-                    completion(loadedData)
-                }
+        var lastRequestTime = Date()
+            guard var components = URLComponents(string: urlString) else {
+                completion(nil)
+                return
+            }
+            components.queryItems = parameters.map { (key, value) in
+                URLQueryItem(name: key, value: value)
             }
-            guard error == nil,
-                  let data = data else {
-                executeCompletionOnMain(nil)
+            components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
+            guard let url = components.url else {
+                completion(nil)
                 return
             }
-            let result = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
-            executeCompletionOnMain(result)
+            var urlRequest = URLRequest(url: url)
```

Comment:
Try not to do all this job (creating URL-components, creating URL, creating URLRequest, creating task, etc.) at all if the last request was done less than 12.5ms ago.
- Create a full copy of the original `get` method and rename it to `executeNetworkGet`
- inside a `get` method:
-- Check that a timing is appropriate for a network request
--- If `yes`, just call `executeNetworkGet`
--- If `no`,  execute `DispatchQueue.global().asyncAfter`, call the original `get` method with the same arguments after 12.5s